File: upsideCakes/Models/Pagamento.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace upsideCakes.Models
{
    public class Pagamento
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int id { get; set; }
        
        public DateOnly data { get; set; }
        public float valor { get; set; }
        public string? formaDePagamento { get; set; }
        public string? nomeCliente { get; set; }

/*
        [ForeignKey("idCliente")]
        public Cliente? cliente { get; set; } // Propriedade de navegação para Cliente
        public int idCliente { get; set; }    

        [ForeignKey("idPedido")]
        public Pedido? pedido { get; set; } // Propriedade de navegação para Pedido
        public int idPedido { get; set; }*/
    }
}

File: upsideCakes/AppUpsideCakes/src/app/models/Pagamento.ts
import { Cliente } from "./Cliente";
import { Pedido } from "./Pedido";
export class Pagamento{

    id: Number = 0;
    data: Date = new Date('2023-11-11');
    valor: number | undefined; // Usar 'number' para representar um valor de ponto flutuant
    formaDePagamento: String = '';
    nomeCliente: String = '';
    //  cliente: Cliente | undefined; //inicializa um obj q n é um tipo primitivo
    //pedido: Pedido | undefined;

}
